master/internal/command/command_manager.go

```diff
@@ -1,81 +1,119 @@
-				resp.Commands = append(resp.Commands, typed)
-			}
+	resp := &apiv1.GetCommandsResponse{}
+	cmds, users, userIDs := cs.listByType(req.Users, req.UserIds, model.TaskTypeCommand)
+	for _, c := range cmds {
+		cmd := c.toCommand()
+		// skip if it doesn't match the requested workspaceID if any.
+		if req.WorkspaceId != 0 && req.WorkspaceId != cmd.WorkspaceId {
+			continue
 		}
-		ctx.Respond(resp)
-
-	case *apiv1.DeleteWorkspaceRequest:
-		ctx.TellAll(msg, ctx.Children()...)
-
-	case *CreateGeneric:
-		taskID := model.NewTaskID()
-		jobID := model.NewJobID()
-		msg.Spec.CommandID = string(taskID)
-		if err := createGenericCommandActor(
-			ctx, c.db, c.rm, taskID, model.TaskTypeCommand, jobID,
-			model.JobTypeCommand, msg.Spec, msg.ContextDirectory,
-		); err != nil {
-			ctx.Log().WithError(err).Error("failed to launch command")
-			ctx.Respond(err)
-		} else {
-			ctx.Respond(taskID)
+		if (len(users) == 0 && len(userIDs) == 0) || users[cmd.Username] || userIDs[cmd.UserId] {
+			resp.Commands = append(resp.Commands, cmd)
 		}
+	}
+	return resp, nil
+}
+
+// GetCommand returns the command matching the commandID.
+func (cs *commandService) GetCommand(req *apiv1.GetCommandRequest) (*apiv1.GetCommandResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	c, err := cs.getNTSC(model.TaskID(req.CommandId), model.TaskTypeCommand)
+	if err != nil {
+		return nil, api.NotFoundErrs("command", req.CommandId, true)
+	}
+
+	return &apiv1.GetCommandResponse{
+		Command: c.toCommand(),
+		Config:  protoutils.ToStruct(c.Config),
+	}, nil
+}
+
+// KillCommand marks the command's allocation as killed.
+func (cs *commandService) KillCommand(req *apiv1.KillCommandRequest) (*apiv1.KillCommandResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	c, err := cs.getNTSC(model.TaskID(req.CommandId), model.TaskTypeCommand)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
+	if err != nil {
+		return nil, fmt.Errorf("failed to kill allocation: %w", err)
+	}
+	return &apiv1.KillCommandResponse{Command: c.toCommand()}, nil
+}
+
+// SetCommandPriority sets the command's resource manager group priority.
+func (cs *commandService) SetCommandPriority(
+	req *apiv1.SetCommandPriorityRequest,
+) (*apiv1.SetCommandPriorityResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	c, err := cs.getNTSC(model.TaskID(req.CommandId), model.TaskTypeCommand)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.setNTSCPriority(int(req.Priority), true)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.SetCommandPriorityResponse{Command: c.toCommand()}, nil
+}
 
-	case echo.Context:
-		ctx.Respond(echo.NewHTTPError(http.StatusNotFound, api.ErrAPIRemoved))
+// toCommand() takes a *command from the command service registry & returns a *commandv1.Command.
+func (c *command) toCommand() *commandv1.Command {
```

Comment:
move this method to `command.go`, all the others to `command_service.go` and remove this file?

Reply:
disagree, because the methods in each XYZ_manager.go file take in API requests & return API responses. To me, mixing up the command_service methods with API-related methods doesn't make as much sense as separating it out.

Reply:
yep i agree, i just mixed up `commandService` and `CommandService`.. kinda confusing names

Reply:
oh well, i do this this should be in `command.go` though. why would a method on command not be in the file the struct is in? it doesn't have to but it is confusing to me.